### PR TITLE
feat: UX polish & multi-device launch

### LINKED ISSUE #31: [Nice To Have] - Jump to end log line

### PR DESCRIPTION
**Pull request by edTheGuy00**
A batch of TUI quality-of-life improvements. 

**What changed:**

- **Launch on multiple devices at once.** Check several connected devices in the new-session dialog and start them all with one Enter. `Space` toggles a device, `a` selects/clears all. If you don't check anything, it works exactly like before.

- **Better "I'm working" feedback.** Status labels shimmer while something's in flight, and spinners now show up during device discovery and launch instead of frozen text.

- **Honest launch states.** Sessions now show `Preparing` / `Launching` while the app is actually building, and only flip to `Running` once it's really up — with live build progress next to the label.

- **Jump back to the latest logs.** When you scroll up and logs keep coming, a "↓ N new — G to jump" hint appears. Press `G` to snap back to the live tail.

- **Only show devices that can actually run.** Devices Flutter can't launch for this project are filtered out, with a clear message if none are left.

- **Little touches.** A green flash on the header when a hot reload lands, and cleaner spacing in the multi-session header.

Closes #31

### CODE REVIEW

File: crates/fdemon-app/src/handler/new_session/launch_context.rs [hunk 1]
```diff
@@ -460,192 +460,322 @@ pub fn handle_entry_point_selected(state: &mut AppState, selected: Option<String
...
+    }
+
+    // Fail fast: resolve the Flutter SDK before creating any session so that
+    // a missing SDK never leaves an orphaned Initializing session in the
+    // manager. flutter_executable() is state-global — the result is the same
+    // for every device in the fan-out loop.
+    //
+    // Note: the SpawnPreAppSources branch (below) does not consume `flutter`,
+    // so resolving it here does not change that branch's behaviour.
+    let flutter_opt = state.flutter_executable();
```
[the remark this answers is not in the document]
Comment (edTheGuy00) [reply crates/fdemon-app/src/handler/new_session/launch_context.rs]: Good catch — valid. The `SpawnPreAppSources` branch created the session without ever checking the SDK, and `PreAppSourcesReady` returned `none()` on a missing SDK without removing it, leaving the session stuck in `Preparing` and consuming a slot.

Fixed in cb652124: `spawn_one` now resolves and guards the Flutter SDK **before** creating the session, so both the direct and pre-app-gated paths fail fast (the device is surfaced as skipped) instead of orphaning a session. Added defense-in-depth in `PreAppSourcesReady` (removes the session + toasts on the now-unreachable missing-SDK arm) and a regression test (`test_handle_launch_pre_app_no_sdk_creates_no_orphan_session`) asserting zero sessions remain.

File: crates/fdemon-app/src/config/launch.rs [hunk 2]
```diff
@@ -330,8 +339,19 @@ impl LaunchConfig {
...
 
-        // Add extra args
-        args.extend(self.extra_args.clone());
+        // Add extra args (validated): must start with '-', contain no NUL bytes,
+        // and stay within the bounded length cap. Entries that fail any check are
+        // silently dropped with a warning — we never reject the whole config for a
+        // single malformed arg, and there is no shell-evaluation risk because args
+        // reach `Command::args()` as separate, non-shell-evaluated elements.
+        for arg in &self.extra_args {
+            let ok = arg.starts_with('-') && !arg.contains('\0') && arg.len() <= MAX_EXTRA_ARG_LEN;
```
[the remark this answers is not in the document]
Comment (edTheGuy00) [reply crates/fdemon-app/src/config/launch.rs]: Valid — confirmed. `parse_tool_args` pushes value tokens of split flag/value pairs verbatim into `extra_args`, so `["--web-port", "8080"]` had `8080` dropped for not starting with `-`, producing a malformed `--web-port` with no value.

Fixed in cb652124: `build_flutter_args` now validates positionally — a token following a flag that lacks an inline `=value` is accepted as that flag's value, regardless of leading `-`. Stray leading positionals are still dropped, and NUL/length checks still apply to every retained token. Added tests `test_extra_args_split_flag_value_preserved`, `test_extra_args_inline_value_does_not_consume_next`, and `test_extra_args_value_token_still_length_checked`.

File: crates/fdemon-app/src/new_session_dialog/target_selector_state.rs [hunk 3]
```diff
@@ -233,6 +239,15 @@ impl TargetSelectorState {
...
         self.scroll_offset = 0; // Reset scroll when devices change
 
+        // Prune checked ids that are no longer present in the new device list.
+        let present: std::collections::HashSet<&str> = self
+            .connected_devices
+            .iter()
+            .map(|d| d.id.as_str())
+            .collect();
+        self.checked_device_ids
+            .retain(|id| present.contains(id.as_str()));
```
[the remark this answers is not in the document]
Comment (edTheGuy00) [reply crates/fdemon-app/src/new_session_dialog/target_selector_state.rs]: Valid — confirmed. Pruning only removed absent ids, so an id still present but flipped to `is_supported == false` after a refresh survived: excluded from `checked_devices()` but still counted by `checked_count()`, giving a stale "N selected".

Fixed in cb652124: `set_connected_devices` now retains a checked id only when its device is **present and supported**, so `checked_count()` and `checked_devices()` agree. Added test `refresh_flipping_unsupported_prunes_checked_count`.
